web/app/scripts/src/directives/proportion-field.es.js

```diff
@@ -0,0 +1,55 @@
+import { angular } from '../libraries';
+
+import ProportionField from '../services/proportion-field';
+
+const moduleProperties = {
+    moduleName: 'incuna-surveys.proportion-field-directive'
+};
+
+const module = angular.module(moduleProperties.moduleName, [
+    ProportionField.moduleName
+]);
+
+module.directive('proportionField', [
+    ProportionField.componentName,
+    function (ProportionField) {
+        return {
+            restrict: 'A',
+            scope: {
+                model: '=',
+                options: '=proportionField'
+            },
+            templateUrl: 'templates/incuna-surveys/form/proportion-field.html',
+            link: function (scope) {
+                scope.fields = [];
+                scope.$watch('options', (options) => {
```

Comment:
Would be nice to provide a wrapper for that, if the explicit action is to only do things with `options` when it's available and then never again.